### ISSUE TITLE
onRemovedFromRoom throwing error - unhandled Exception: type '_InternalLinkedHashMap<Object?, Object?>' is not a subtype of type 'HMSPeer' in type cast

### ISSUE THREAD

**Issue by abbas2501** [100mslive/100ms-flutter#207]
This issue lies in the 3rd line in following code snippet, map value needs to be passed thru the factory method fromMap in order to type cast properly

```
factory HMSPeerRemovedFromPeer.fromMap(Map map) {
    return HMSPeerRemovedFromPeer(
      peerWhoRemoved: map['peer_who_removed'] as HMSPeer,
      reason: map['reason'] as String,
      roomWasEnded: map['room_was_ended'] as bool,
    );
  }
```

**Comment by ygit** [comment]
@abbas2501 Ensure that you are using the latest version of both the [Flutter](https://docs.100ms.live/flutter/v2/release-notes/release-notes) & underlying [Android SDK](https://docs.100ms.live/android/v2/release-notes/Release-Notes).

[Flutter - 0.4.0](https://docs.100ms.live/flutter/v2/release-notes/release-notes#0-4-0-2021-10-22)
[Android - 2.1.8](https://docs.100ms.live/android/v2/release-notes/Release-Notes#2-1-8-2021-11-03)

**Comment by abbas2501** [comment]
> @abbas2501 Ensure that you are using the latest version of both the [Flutter](https://docs.100ms.live/flutter/v2/release-notes/release-notes) & underlying [Android SDK](https://docs.100ms.live/android/v2/release-notes/Release-Notes).
> 
> [Flutter - 0.4.0](https://docs.100ms.live/flutter/v2/release-notes/release-notes#0-4-0-2021-10-22) [Android - 2.1.8](https://docs.100ms.live/android/v2/release-notes/Release-Notes#2-1-8-2021-11-03)

The issue still exists on flutter 0.4.0 as I can see currently. Have updated to 2.1.8 android also now.

**Comment by ygit** [comment]
@abbas2501 you are facing the issue only on android or iOS as well?

**Comment by abbas2501** [comment]
Have tested only on android, but this clearly is a flutter issue, isn't it ? We are basically trying type casting a map to an object of type HMSPeer

**Comment by abbas2501** [comment]
> @abbas2501 you are facing the issue only on android or iOS as well?

Please look for this code snippet in the sdk. I have fixed it for myself and using it as a localy sdk. Just informed here so it can be fixed for everyone.

**Comment by ygit** [comment]
Sure

**Comment by ygit** [comment]
@abbas2501 We have resolved the issue & will release it soon

**Comment by ygit** [comment]
Resolved in version 0.5.0
